modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/web/internal/portlet/action/test/ObjectDefinitionExportImportTest.java

```diff
@@ -65,6 +81,128 @@ public void testExportImportObjectDefinition() throws Exception {
 			"test-object-definition.portuguese-locale.json",
 			"test-object-definition.site-default-locale.json",
 			"TESTOBJECTDEFINITIONPORTUGUESE", "TestObjectDefinitionPortuguese");
+
+		// Import published object definition
+
+		String publishedObjectDefinitionJSONString =
+			_jsonFactory.createJSONObject(
+				_defaultObjectDefinitionJSONString
+			).put(
+				"active", true
+			).put(
+				"externalReferenceCode", "TESTOBJECTDEFINITION1"
+			).put(
+				"name", "TestObjectDefinition1"
+			).put(
+				"status",
+				_jsonFactory.createJSONObject(
+				).put(
+					"code", 0
+				).put(
+					"label", "approved"
+				).put(
+					"label_i18n", "Approved"
+				)
+			).toString();
+
+		testExportImportJSONString(
+			publishedObjectDefinitionJSONString,
+			publishedObjectDefinitionJSONString, "TESTOBJECTDEFINITION1",
+			"TestObjectDefinition1");
+
+		// Cannot modify the status of published object definition
+
+		Class<?> clazz = getClass();
+
+		testFailedImportJSONString(
+			_jsonFactory.createJSONObject(
+				publishedObjectDefinitionJSONString
+			).put(
+				"status",
+				_jsonFactory.createJSONObject(
+				).put(
+					"code", 2
+				).put(
+					"label", "draft"
+				).put(
+					"label_i18n", "Draft"
+				)
+			).toString(),
+			StringUtil.read(
+				clazz.getResourceAsStream(
+					"dependencies/test-object-definition.draft.error-message." +
+						"json")),
+			"TESTOBJECTDEFINITION1", "TestObjectDefinition1");
```

Comment:
I think the JSON file name could be: test-object-definition.status-error-message.json
what do you think?

Reply:
The exception thrown is ObjectDefinitionStatusException.